src/util/index.ts

```diff
@@ -85,6 +91,8 @@ export function updateAvailableDepositAndForeclosureTime(
   patron.patronTokenCostScaledNumerator = steward.totalPatronOwnedTokenCost(
     patron.address as Address
   );
+  patron.availableDeposit = steward.depositAbleToWithdraw(tokenPatron);
+  patron.foreclosureTime = getForeclosureTimeSafe(steward, tokenPatron);
```

Comment:
Updating (to the new) foreclosure has to happen after calculating `heldUntil`

Reply:
Good catch